app/views/components/submit_button.scala.html

```diff
@@ -14,8 +14,8 @@
  * limitations under the License.
  *@
 
-@(messageKey: String = "site.save_and_continue")(implicit messages: Messages)
+@(message: Html)
```

Comment:
Why a messages is `Html` not `String`?

Reply:
Agreed, I see no need for this to be a `Html`

Reply:
We could just do this as
```
@(text: String = "")(implicit messages: Messages)
<div class="section">
   <button id="submit" class="button">
        @if(text.nonEmpty) text else @messages("site.save_and_continue")) 
   </button>
</div>
```

Reply:
Still passing messages - this makes controller as dummy as possible without any logic inside

Reply:
Agreed.  Have already suggested that.     I think this whole PR is overly complicated and that components that effectively render "static" text should be free to do their own message look-up.

The rule about supplying components with message strings rather than keys should only apply where components render different messages.

Reply:
Moreover I think it is not responsibility of component to provide to itself content

Reply:
IMO there shouldn't be a default value for it. It's more common for us to use Continue etc not always Save and continue. I'll change it to String, but without any dependency with messages and without default value